diff --git a/ios/Utils/URL+MediaLibraryAssets.swift b/ios/Utils/URL+MediaLibraryAssets.swift
index c30c7cab2772145bea9f7000e20d89a7f1f0d390..1cb8900630868a5d05e1db138c10454dadb6d3b0 100644
--- a/ios/Utils/URL+MediaLibraryAssets.swift
+++ b/ios/Utils/URL+MediaLibraryAssets.swift
@@ -58,6 +58,7 @@ private extension PHImageManager {
 
         if let error = info?[PHImageErrorKey] as? Error {
           continuation.resume(throwing: PlayerItemLoadException("Failed to request an AVAsset for \(urlString): \(error.localizedDescription)"))
+          return
         }
 
         continuation.resume(returning: asset)
